src/providers/mbed_crypto/key_management.rs

```diff
@@ -99,29 +98,29 @@ impl Provider {
         let key_name = op.key_name;
         let key_attributes = op.attributes;
         let key_triple = KeyTriple::new(app_name, ProviderID::MbedCrypto, key_name);
+
         let mut store_handle = self
             .key_info_store
             .write()
             .expect("Key store lock poisoned");
-        if key_info_exists(&key_triple, &*store_handle)? {
-            return Err(ResponseStatus::PsaErrorAlreadyExists);
-        }
-        let key_id = create_key_id(
-            key_triple.clone(),
-            key_attributes,
-            &mut *store_handle,
-            &self.id_counter,
-        )?;
+
+        store_handle.already_exists(&key_triple)?;
+
+        let key_id = create_key_id(&self.id_counter)?;
 
         let _guard = self
             .key_handle_mutex
             .lock()
             .expect("Grabbing key handle mutex failed");
 
         match psa_crypto_key_management::generate(key_attributes, Some(key_id)) {
-            Ok(_) => Ok(psa_generate_key::Result {}),
+            Ok(_) => {
+                // If this fails, then we will have a zombie key in the Mbed Crypto backend. This
+                // is probably not a big problem apart from memory consumption.
```

Comment:
It can still cause some operations to fail: if you restart the service and don't have a higher ID used (the IDs are allocated incrementally, right?) then you'll end up trying to use this one and that will fail, but the next generate/import will be fine, so it just needs a retry.

Reply:
ah yes you are correct, I guess this is the problem with this method... And will be the same with PKCS11 even though we use a random key ID generator: if it happens that we create the same key ID of a zombie key then the operation will have to be retried. In the worst case, if that happens a lot, one would have to re-try the key generation several times before finding an unallocated key ID.
Should we try to delete the key then, if insert_key_id fails? Here and for the PKCS11 provider.